Development documentation

kerimoyle opened the pull request:
I'm not sure where the development documentation will be stored in the future?  The generated XML from these files is already in the website repo.

kerimoyle committed:
Update dev_setup.rst

Making the link to Windows installer for libxml2 more obvious.

kerimoyle committed:
Update dev_building.rst

kerimoyle committed:
WIP updating windows instructions

kerimoyle committed:
WIP updating dev instructions for windows

kerimoyle committed:
WIP

kerimoyle committed:
Create libCellMLBuilding-CMakeOptionsInRed.png

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
WIP - including tables of options

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Merge branch 'documentation' of https://github.com/kerimoyle/libcellml-keri into documentation

kerimoyle committed:
Delete dev_building_newer.rst

kerimoyle committed:
Update dev_building_newer.rst

kerimoyle committed:
Merge branch 'documentation' of https://github.com/kerimoyle/libcellml-keri into documentation

kerimoyle committed:
Create dev_building2.rst

kerimoyle committed:
WIP adding figures and testing

kerimoyle committed:
WIP

kerimoyle committed:
Update dev_building2.rst

kerimoyle committed:
v3

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Adding sphinx instructions

kerimoyle committed:
Update dev_setup.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Update dev_building3.rst

kerimoyle committed:
Moving to main building page

nickerso committed:
Starting work on updating the developer documentation for the project.

This incomplete work is aimed at improving the documentation overall, mostly with a focus on helping potential contributors get this up and running. I have also added some placeholder text for when we have installers ready for users.

nickerso committed:
adding links to moved developer documents.

kerimoyle committed:
Update dev_setup.rst

kerimoyle committed:
Added python links to requirements for setup

kerimoyle committed:
starting to write glossary terms and links

kerimoyle committed:
Getting refs to work

kerimoyle committed:
One sentence per line

kerimoyle committed:
Glossary tweaks

I'm not sure how best to format the glossary page: If there is more than one line (because of the one sentence per line thing) then the first sentence is rendered in bold and looks horrible ...

kerimoyle committed:
WIP playing - only committing to switch

The css tag for the first line underneath a definition is set as <dt> which makes it the start of a new list, and therefore title case.  It doesn't seem to happen in the readthedocs version so might be something different between my build and theirs which affects formatting.

nickerso committed:
Merge remote-tracking branch 'keri/documentation' into doc-updates

nickerso committed:
Merge branch 'develop' into doc-updates

nickerso committed:
Clean and tidy following restructure.

Updating image paths for new locations, correcting some internal cross references, trying to remove all the sphinx-build warnings.

kerimoyle committed:
Merge branch 'develop' into doc-updates

kerimoyle committed:
Merge pull request #63 from cellml/develop

Update from upstream: segfault in flattener

hsorby committed:
Merge in changes from 'prime/develop' fixing conflicts.

kerimoyle committed:
Merge pull request #64 from cellml/develop

Update from upstream

kerimoyle committed:
update dev docs

kerimoyle committed:
Updated docs

kerimoyle committed:
Update dev_building.rst

kerimoyle committed:
Update dev_coding_standards.rst

kerimoyle committed:
Merge branch 'develop' into development-documentation

kerimoyle committed:
Merge pull request #65 from cellml/develop

Update from upstream

kerimoyle committed:
REanmed conf.py -> conf.py.in

hsorby commented on docs/dev_coding_standards.rst:
```diff
@@ -0,0 +1,41 @@
+Variable naming
+===============
+
+**TODO** This makes no sense ... !!
```
It does make sense for the most part they are examples on how variables should be named which is different to how the google style guide suggests it should be done.

kerimoyle replied:
What is the example?? Camel case?  We've said that already!  Give me an actual example from within libcellml and it might make sense, but as is this tells me nothing ...

hsorby commented on docs/dev_configuration_options.rst:
```diff
@@ -1,27 +1,29 @@
+:code:`UNIT_TESTS`        ON           Enable tests.
+======================== ============ ===========================================
 
-:sup:`*` In CMake GUI Configuration applications this option is given in full ``LIBCELLML_TREAT_WARNINGS_AS_ERRORS``
```
I think `TWAE` needs a footnote or some sort of special consideration and not ignored.

kerimoyle replied:
I don't understand.  The gui config is shown elsewhere where it's spelled out in full?

hsorby commented on docs/dev_contribution.rst:
```diff
@@ -55,89 +57,85 @@ General labels currently used for libCellML are:
 Refactoring means improving the code without adding features, and the tests provide validation that the refactored code performs as well as before.
 
 For simple or obvious bugs, which have fallen through the testing gaps, just the implementation is fine.
+**TODO** KRM I don't understand this sentence?
```
It says for simple or (now) obvious bugs you don't need to write a test just fix the implementation.

kerimoyle replied:
Doesn't that go against all the testing rationale?  I thought that was kinda the point - that the tests could be relied upon to test whatever bugs we knew about?

hsorby commented on docs/dev_review.rst:
```diff
@@ -1,51 +1,56 @@
+Read the documentation
 ======================
 
+**TODO** Not sure any of this is current??
```
It is.

kerimoyle replied:
Then it doesn't make sense to me?  Here's what the link is pointing to: 
![image](https://user-images.githubusercontent.com/46887220/107562511-cb573c80-6c44-11eb-9e91-11d8ca73a85b.png)
What does this refer to? 
>> The Documentation Builder link (entry 2 in step 5) will take you to the build for the documentation.
On this page, you can see the steps taken to build the documentation.
In the last step of the build (step 7), there is a link 'dox' (entry 2) that will take you to the built documentation.

hsorby commented on docs/dev_review.rst:
```diff
@@ -1,51 +1,56 @@
+The "Details" link at the bottom of the pull request web page will take you to the build records of the library.
+This page shows the results of the unit tests for each target operating system, the results of the coverage test, the results of the memory check test, and the results of the documentation build.
+
+The coverage statistics for the library (generated from :gcovr:`gcovr<>` ) can be reached through the :ref:`Coverage Statistics <coverage_statistics>` page **TODO** is this still true??
```
Yes.

kerimoyle replied:
Two problems there:
- Don't we always have 100% API coverage as you can't merge without it?  
- And did you look at that page?? This is what it says, which doesn't seem entirely complete somehow!

**TODO**
The output from the coverage testing using gcov is available TODO.

.. note::

   The coverage test sometimes (this behaviour has been observed on `macOS <https://en.wikipedia.org/wiki/MacOS>`_ using `Clang <https://clang.llvm.org/>`_) reports single lines containing only a closing curly brace as not covered.
   This is currently being treated as a false positive.
   This can be seen in the 'Missing' column of the test report where only single lines are reported.
   For the case discussed here, the reported line should only contain a single closing curly brace.
   In this situation, we will accept the missed coverage report.

hsorby commented on docs/dev_coding_standards.rst:
```diff
@@ -0,0 +1,41 @@
+.. _dev_coding_standards:
```
For me it is a single standard so the plurality here doesn't hold.

kerimoyle replied:
Well, you've said that we use google plus some tweaks ... so that makes more than one ;) And just as a title, "standard" doesn't really read well.

kerimoyle commented on docs/coverage_statistics.rst:
```diff
@@ -1,13 +1,14 @@
-The output from the coverage testing using gcov is available `here <coverage/index.html>`_.
+**TODO**
+The output from the coverage testing using gcov is available TODO.
 
```
Need to have a link to where the coverage testing actually is

nickerso replied:
this will need to be set to be the location once the coverage stats are getting pushed to the website...

kerimoyle commented on docs/coverage_statistics.rst:
```diff
@@ -1,13 +1,14 @@
 
-   The coverage testing pages are **not** currently available on `Read the Docs <http://libcellml.readthedocs.io/en/latest/>`_.
    The coverage test sometimes (this behaviour has been observed on `macOS <https://en.wikipedia.org/wiki/MacOS>`_ using `Clang <https://clang.llvm.org/>`_) reports single lines containing only a closing curly brace as not covered.
    This is currently being treated as a false positive.
```
Remove links here

kerimoyle commented on docs/current_thinking.rst:
```diff
@@ -14,11 +14,11 @@ This document simply outlines some of the current rationale that has an influenc
   - No avenue to retrieve remote external references.
 
```
This entire document needs to be updated!

kerimoyle commented on docs/dev_building.rst:
```diff
@@ -1,105 +1,187 @@
 
   cd $LIBCELLML_BUILD
-  cmake -DBUILD_TYPE=Release $LIBCELLML_SRC
+  cmake -DBUILD_TYPERelease $LIBCELLML_SRC
```
Think this is a typo? still needs the equals?

kerimoyle commented on docs/dev_building.rst:
```diff
@@ -1,105 +1,187 @@
+For Windows only the location of the libXML2 library must be specified through the command line by adding the parameter::
 
-  -DLibXml2_DIR="C:\Program Files\libxml2 2.9.6\lib\cmake"
+  -DLibXml2_DIR "C:\Program Files\libxml2 2.9.6\lib\cmake"
```
Also missing an equals??

kerimoyle commented on docs/dev_building.rst:
```diff
@@ -1,105 +1,187 @@
+This library is built into Linux and MacOS systems, so this step is only needed on Windows.
+
+This assumes that the recommended LibXml2 binaries have been installed to the default location :code:`C:\Program Files\libxml2 2.9.6`.  
+Please note that libCellML will only work with a 64-bit installation of libXML2.  
```
Put 64bit info in an NB block.

kerimoyle commented on docs/dev_building.rst:
```diff
@@ -1,105 +1,187 @@
+
+Could NOT find Sphinx (missing: SPHINX_EXECUTABLE)
+--------------------------------------------------
+`Sphinx <http://www.sphinx-doc.org/en/master/>`_ is a documentation generator used by libCellML to collect documentation on Python bindings, which requires Python 3.5.2 or later in order to run.  
```
Make proper external links

kerimoyle committed:
Update dev_building.rst

kerimoyle commented on docs/dev_contribution.rst:
```diff
@@ -1,50 +1,52 @@
-General labels currently used for libCellML are:
+A Github issue may be assigned labels by the project administrators to help identify its status at a glance.
+General labels currently used for libCellML are: **TODO These are not current?**
 
```
These labels don't line up with the ones we've got, which are: 
![image](https://user-images.githubusercontent.com/46887220/107565471-a5339b80-6c48-11eb-9152-989670ea1faa.png)

kerimoyle commented on docs/dev_setup.rst:
```diff
@@ -1,138 +1,178 @@
+  The *Community* edition is more than sufficient for the needs of libCellML.
+  To minimise the size of the installation, you may install only the C++ compiler.
+  This component (and its requirements) is sufficient for building libCellML.
+- **Linux** TODO
```
Needs info

kerimoyle commented on docs/dev_setup.rst:
```diff
@@ -1,138 +1,178 @@
+  To minimise the size of the installation, you may install only the C++ compiler.
+  This component (and its requirements) is sufficient for building libCellML.
+- **Linux** TODO
+- **macOS** TODO
```
Needs info

kerimoyle commented on docs/dev_testing_code.rst:
```diff
@@ -1,16 +1,39 @@
+     The test files can then be run individually as needed using :code:`python test_classname.py`, where :code:`classname` is the lowercase name of the class to test.
+- **Coverage tests** Coverage tests make sure that each line of code is covered by one or more unit tests.
+  - A summary of coverage of the C++ code is accessible through :code:`make llvm_coverage`, which result in statistics of line coverage.
+  - For detailed information regarding specific lines in each file that need test coverage, use the command **TODO**.
```
Needs info

kerimoyle commented on docs/dev_testing_code.rst:
```diff
@@ -1,16 +1,39 @@
+  - A summary of coverage of the C++ code is accessible through :code:`make llvm_coverage`, which result in statistics of line coverage.
+  - For detailed information regarding specific lines in each file that need test coverage, use the command **TODO**.
+- **Formatting tests** Check that the code is correctly formatted using the :code:`make test_clang_format`, if you have CLang installed.  
+  **TODO** Check what other OS use?
```
Needs info

kerimoyle commented on docs/dev_testing_code.rst:
```diff
@@ -1,16 +1,39 @@
+  - For detailed information regarding specific lines in each file that need test coverage, use the command **TODO**.
+- **Formatting tests** Check that the code is correctly formatted using the :code:`make test_clang_format`, if you have CLang installed.  
+  **TODO** Check what other OS use?
+- **Memory tests** TODO
```
Needs info

kerimoyle commented on docs/dev_versions.rst:
```diff
@@ -0,0 +1,13 @@
+
+This document outlines the versions of packages which libCellML requires.  
+
+**TODO**
```
Needs lots of info ...

kerimoyle committed:
tweaks

kerimoyle committed:
Rename development documentation files

kerimoyle committed:
Merge branch 'develop' into development-documentation

kerimoyle committed:
rename conf.py -> conf.py.in

kerimoyle committed:
rename glossary so it doesn't conflict ids

kerimoyle committed:
Change name of glossary as id is taken from heading

kerimoyle committed:
updated figure names for captions

kerimoyle committed:
Update setup.rst

nickerso committed:
Starting to update roadmap document

hsorby committed:
Update roadmap.rst

nickerso committed:
Merge pull request #6 from hsorby/patch-1

Update roadmap.rst

nickerso committed:
Merge branch 'main' into roadmap-update

nickerso committed:
Merge branch 'roadmap-update' of https://github.com/nickerso/libcellml into roadmap-update

nickerso committed:
Update README with minor tweaks

Updating README to now point to libcellml.org rather than readthedocs and to bring things into line with the current status.

nickerso committed:
fixing broken buildbot links

nickerso committed:
Merge branch 'main' into development-documentation

nickerso committed:
Merge branch 'development-documentation' of https://github.com/kerimoyle/libcellml into roadmap-update

nickerso committed:
Update coding_standards.rst

nickerso committed:
Update contributing.rst

nickerso committed:
Update coverage_statistics.rst

nickerso committed:
Update review_process.rst

nickerso reviewed (dismissed):
Matches the updates already deployed on libcellml.org, with a few small modifications. Gets us to a good point to move forward with gradual updates.

nickerso committed:
Update README.rst

Removing the note on alpha status of libCellML.
